javagen/src/main/java/com/azure/autorest/model/clientmodel/ClassType.java

```diff
@@ -15,14 +15,68 @@
 import java.time.OffsetDateTime;
 import java.time.ZoneOffset;
 import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
 
 /**
  * The details of a class type that is used by a client.
  */
 public class ClassType implements IType {
+
+
+    private static class ClassDetails {
+
+        private Class<?> azureClass;
+        private Class<?> genericClass;
+        public ClassDetails(Class<?> azureClass, Class<?> genericClass) {
+            this.azureClass = azureClass;
+            this.genericClass = genericClass;
+        }
+
+        public Class<?> getAzureClass() {
+            return azureClass;
+        }
+
+        public Class<?> getGenericClass() {
+            return genericClass;
+        }
+
+    }
+
+    public static final Map<Class<?>, ClassDetails> classTypeMapping = new HashMap<Class<?>, ClassDetails>() {{
+        put(com.azure.core.http.rest.RestProxy.class, new ClassDetails(com.azure.core.http.rest.RestProxy.class, com.generic.core.http.RestProxy.class));
```

Comment:
We probably not able to do `com.generic.core.http.RestProxy.class` before generic-core get released somewhere.